docs/advanced/featureflags.md

```diff
@@ -0,0 +1,280 @@
+
+:::note
+`ItemStack` has a speciality `isItemEnabled(FeatureFlagSet)` method, This is so that empty stacks are treated as enabled even if the required features for backing `Item` does not match the enabled features.
+
+It is recommended to make use of this speciality over `Item#isEnabled` where possible.
+:::
+
+```java
+requiredFeatures.isSubsetOf(enabledFeatures);
+featureElement.isEnabled(enabledFeatures);
+itemStack.isItemEnabled(enabledFeatures);
+```
+
+### Feature Packs
+
+Feature Packs are a type of pack which not only loads resources and/or data but also has the ability to toggle on a given set of Feature Flags.
+
+These flags are defined in the `pack.mcmeta` JSON file at the root of this pack, which follows the below format.
+```json5
+{
+    "features": [
+        "enabled": [
+            // Identifier of a Feature Flag to be enabled
+            // Must be a valid registered flag
+            "examplemod:experimental"
+        ]
+    ]
+}
+```
+
+There are a couple methods in which you can provide a feature pack to your users.
+
+Installing these packs differs depending on if its being installed into a single player world or a server.
+
+**External**
+
+External packs are provided to your users in datapack form.
+
+**Installing into a single player world**
+1. Create a new world.
+2. Navigate to the datapack selection screen.
+3. Drag and drop the datapack zip file onto game window.
+4. Move the newly available datapack over to the `Selected` packs list.
+5. Confirm changes by clicking `Done`.
+
+The game will now warn you about any newly selected experimental features, potential bugs, issues and crashes.
+You can confirm these changes by clicking `Proceed` or `Details` to see an extensive list of all selected packs and which features they would enable.
+
+:::note
+External Feature Packs do not show up in the Experiments screen.
+
+To disable these packs after enabling them, navigate back into the datapacks screen and move the external packs back into `Available` from `Selected`.
+
+The Experiments screen only allows toggling on or off Built-In Feature Packs.
+:::
+
+**Installing into a server**
+
+Enabling Feature Packs can only be done during initial world creation, and cannot be disabled once enabled.
+
+1. Create the directory `./world/datapacks`
+2. Upload the datapack zip file into the newly created directory
+3. Open your servers `server.properties` file
+4. Add the datapack zip file name (excluding `.zip`) to `initial-enabled-packs` (separating each pack by a `,`)
+
+The Feature Pack is now installed and enabled
+
+**Built-In**
+
+Built-In packs are bundled with your mod and are made available to the game using the `AddPackFindersEvent` event.
+```java
+@SubscribeEvent
+public static void addFeaturePacks(final AddPackFindersEvent event) {
+    event.addPackFinders(
+            // 'packLocation': Path relative to your mods 'resources' pointing towards this pack
```

Comment:
The parameter name should not be added to the comment, an explanation of what is expected here should be enough.